### PR TITLE
feat(linter): update @typescript-eslint/* package versions

### PR DESCRIPTION
**Pull request by leosvelperez**
Updates the `@typescript-eslint/*` packages in preparation for supporting TypeScript 5.3, which will be supported once the [Angular v17.1 support](https://github.com/nrwl/nx/pull/20556) lands.

## Related Issue(s)

Fixes #

### CODE REVIEW

**Review by mandarini — APPROVED** [review]

File: packages/eslint/migrations.json [hunk 1]
```diff
@@ -108,10 +108,19 @@
       }
     },
     "17.2.0": {
```
Comment (meeroslav) [inline packages/eslint/migrations.json:110]: 17.2.0 is out so you might want to change this to 17.3.0 or 17.2.1

**Review by meeroslav — APPROVED** [review]
Left one minor comment, but otherwise it's good

File: packages/vue/src/generators/library/__snapshots__/library.spec.ts.snap [hunk 2]
```diff
@@ -110,8 +110,8 @@ exports[`lib should add vue, vite and vitest to package.json 1`] = `
     "@nx/vue": "0.0.1",
     "@swc-node/register": "~1.6.7",
     "@swc/core": "~1.3.85",
-    "@typescript-eslint/eslint-plugin": "^6.9.1",
-    "@typescript-eslint/parser": "^6.9.1",
+    "@typescript-eslint/eslint-plugin": "^6.13.2",
+    "@typescript-eslint/parser": "^6.13.2",
```
Comment (AgentEnder) [inline packages/vue/src/generators/library/__snapshots__/library.spec.ts.snap:114]: We should update this test to not use a snapshot - its no fun to have to update it every time we change an unrelated package version. Could be a good follow up PR?

**Review by AgentEnder — APPROVED** [review]

File: packages/vue/src/generators/library/__snapshots__/library.spec.ts.snap (hunk 2 above)
Comment (leosvelperez) [reply packages/vue/src/generators/library/__snapshots__/library.spec.ts.snap:114]: I agree. I mentioned it before to @FrozenPandaz, but it hasn't been a priority. There are multiple tests with too broad assertions.

**Pull request merged**